tests/test_itd_service.rb

```diff
@@ -0,0 +1,291 @@
+                 itd.load_bal_method_start_port)
+    assert_equal(itd.default_load_bal_method_proto,
+                 itd.load_bal_method_proto)
+    itd = lb_helper(load_bal_enable: false)
+    assert_equal(itd.load_bal_enable,
+                 itd.default_load_bal_enable)
+    assert_equal(itd.load_bal_buckets, itd.default_load_bal_buckets)
+    assert_equal(itd.load_bal_mask_pos, itd.default_load_bal_mask_pos)
+    assert_equal(itd.load_bal_method_bundle_hash,
+                 itd.default_load_bal_method_bundle_hash)
+    assert_equal(itd.load_bal_method_bundle_select,
+                 itd.default_load_bal_method_bundle_select)
+    assert_equal(itd.load_bal_method_end_port,
+                 itd.default_load_bal_method_end_port)
+    assert_equal(itd.load_bal_method_start_port,
+                 itd.default_load_bal_method_start_port)
+    assert_equal(itd.load_bal_method_proto,
+                 itd.default_load_bal_method_proto)
+    itd.destroy
+  end
+
+  def test_nat_destination
+    itd = ItdService.new('new_group')
+    if node.product_id =~ /N9/
+      assert_nil(itd.nat_destination)
+      assert_raises(Cisco::UnsupportedError) do
+        itd.nat_destination = false
+      end
+      return
+    end
+    itddg = ItdDeviceGroup.new('abc')
+    ItdDeviceGroupNode.new(itddg.name, '1.1.1.1', 'ip')
+    itd.device_group = 'abc'
+    itd.virtual_ip = ['ip 2.2.2.2 255.255.255.0']
+    intf = interfaces[0].dup
+    ii = [[intf.insert(8, ' '), '2.2.2.2']]
+    itd.ingress_interface = ii
+    itd.nat_destination = true
+    assert_equal(true, itd.nat_destination)
+    itd.nat_destination = itd.default_nat_destination
+    assert_equal(itd.default_nat_destination, itd.nat_destination)
+    itd.destroy
+  end
+
+  def test_shutdown
+    itd = ItdService.new('new_group')
+    itddg = ItdDeviceGroup.new('abc')
+    ItdDeviceGroupNode.new(itddg.name, '1.1.1.1', 'ip')
+    itd.device_group = 'abc'
+    itd.virtual_ip = ['ip 2.2.2.2 255.255.255.0']
+    intf = Interface.new(interfaces[0])
+    intf.switchport_mode = :disabled
+    intf_dup = interfaces[0].dup
+    ii = [[intf_dup.insert(8, ' '), '2.2.2.2']]
+    itd.ingress_interface = ii
+    itd.shutdown = false
+    assert_equal(false, itd.shutdown)
+    itd.shutdown = itd.default_shutdown
+    assert_equal(itd.default_shutdown,
+                 itd.shutdown)
+    itd.destroy
+  end
+
+  def test_peer_vdc
+    itd = ItdService.new('new_group')
+    parray = %w(vdc1 ser1)
+    if node.product_id =~ /N9/
+      assert_nil(itd.peer_vdc)
+      assert_raises(Cisco::UnsupportedError) do
+        itd.peer_vdc = parray
+      end
+      return
+    end
+    itd.peer_vdc = parray
+    assert_equal(parray, itd.peer_vdc)
+    itd.peer_vdc = itd.default_peer_vdc
+    assert_equal(itd.default_peer_vdc,
+                 itd.peer_vdc)
+    itd.destroy
+  end
+
+  def test_peer_local
+    itd = ItdService.new('new_group')
+    service = 'ser1'
+    if node.product_id =~ /N7/
```

Comment:
Please use `validate_property_excluded?` instead of hardcoding the platform.

https://github.com/cisco/cisco-network-node-utils/blob/6a0a58a2bf5627b6c91a97a5ae2abb18e2488cb4/tests/test_portchannel_global.rb#L55